File: C#.Net/JDI/Core/Settings/JDISettings.cs
using System;
using Epam.JDI.Core.Interfaces.Settings;
using Epam.JDI.Core.Logging;
using JDI_Core.Interfaces.Settings;
using static System.Int32;
using static Epam.JDI.Core.Properties.Settings;

// ReSharper disable InconsistentNaming

namespace Epam.JDI.Core.Settings
{
    public class JDISettings
    {
        public static ILogger Logger;
        public static IAssert Asserter;
        public static TimeoutSettings Timeouts = new TimeoutSettings();
        public static bool IsDemoMode;
        public static HighlightSettings HighlightSettings = new HighlightSettings();
        public static bool ShortLogMessagesFormat = true;
        public static string JDISettingsPath = "test.properties";
        public static bool ExceptionThrown;
        public static IDriver<IDisposable> DriverFactory;
        public static bool UseCache;

        public static void ToLog(string message, LogLevels level)
        {
            switch (level)
            {
                case LogLevels.Debug:
                    Logger.Debug(message);
                    return;
                case LogLevels.Error:
                    Logger.Error(message);
                    return;
            }
            Logger.Info(message);
        }

        public static string UseDriver(string driverName)
        {
            return DriverFactory.RegisterDriver(driverName);
        }

        public static void InitFromProperties()
        {
            FillFromSettings(p => DriverFactory.RegisterDriver(p), "Driver");
            FillFromSettings(p => DriverFactory.SetRunType(p), "RunType");
            FillFromSettings(p => Timeouts.WaitElementSec = Parse(p), "TimeoutWaitElement");
            FillFromSettings(p => ShortLogMessagesFormat = p.ToLower().Equals("short"), "LogMessageFormat");
            FillFromSettings(p =>
                UseCache = p.ToLower().Equals("true") || p.ToLower().Equals("1"), "Cache");
            FillFromSettings(p =>
                UseCache = p.ToLower().Equals("true") || p.ToLower().Equals("1"), "DemoMode");
            FillFromSettings(p => HighlightSettings.SetTimeoutInSec(Parse(p)), "DemoDelay");
        }

        protected static void FillFromSettings(Action<string> action, string name)
        {
            //var b = System.Configuration.ConfigurationManager.AppSettings["DriversFolder"];
            //var a = Properties.Settings.Default["DriversFolder"];
            ExceptionUtils.AvoidExceptions(() => action.Invoke(Default[name].ToString()));
        }

        public static void InitFromProperties(string propertyPath)
        {
            JDISettingsPath = propertyPath;
            InitFromProperties();
        }

        public static void NewTest()
        {
            ExceptionThrown = false;
        }

        public static Exception Exception(string msg, Exception ex)
        {
            ExceptionThrown = true;
            return Asserter.Exception(msg, ex);
        }

        public static Exception Exception(string msg)
        {
            ExceptionThrown = true;
            return Asserter.Exception(msg);
        }
    }
}

File: jdi-uitest-core/src/main/java/com/epam/jdi/uitests/core/settings/JDISettings.java
package com.epam.jdi.uitests.core.settings;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

import com.epam.jdi.uitests.core.interfaces.IAsserter;
import com.epam.jdi.uitests.core.interfaces.settings.IDriver;
import com.epam.jdi.uitests.core.logger.ILogger;
import com.epam.jdi.uitests.core.logger.LogLevels;

import java.io.IOException;

import static com.epam.commons.PropertyReader.fillAction;
import static com.epam.jdi.uitests.core.settings.JDIPropertiesReader.getProperties;
import static java.lang.Integer.parseInt;

/**
 * Created by Roman_Iovlev on 6/9/2015.
 */
public abstract class JDISettings {
    public static ILogger logger;
    public static IAsserter asserter;
    public static TimeoutSettings timeouts = new TimeoutSettings();
    public static boolean isDemoMode;
    public static HighlightSettings highlightSettings = new HighlightSettings();
    public static boolean shortLogMessagesFormat = true;
    public static String jdiSettingsPath = "test.properties";
    public static IDriver driverFactory;
    public static boolean USE_CACHE = false;

    protected JDISettings() {
    }

    public static void toLog(String message, LogLevels level) {
        switch (level) {
            case INFO:
                logger.info(message);
                break;
            case ERROR:
                logger.error(message);
                break;
            default:
                logger.debug(message);
        }
    }

    public static String useDriver(String driverName) {
        return driverFactory.registerDriver(driverName);
    }

    public static synchronized void initFromProperties() throws IOException {
        getProperties(jdiSettingsPath);
        fillAction(driverFactory::registerDriver, "driver");
        fillAction(driverFactory::setRunType, "run.type");
        fillAction(p -> shortLogMessagesFormat = p.toLowerCase().equals("short"), "log.message.format");
        fillAction(p -> USE_CACHE =
                p.toLowerCase().equals("true") || p.toLowerCase().equals("1"), "cache");
        fillAction(p -> isDemoMode =
                p.toLowerCase().equals("true") || p.toLowerCase().equals("1"), "demo.mode");
        fillAction(p -> highlightSettings.setTimeoutInSec(parseInt(p)), "demo.delay");
        fillAction(p -> timeouts.setDefaultTimeoutSec(parseInt(p)), "timeout.wait.element");
        // fillAction(p -> timeouts.waitPageLoadSec = parseInt(p), "timeout.wait.pageLoad");
    }

    public static void initFromProperties(String propertyPath) throws IOException {
        jdiSettingsPath = propertyPath;
        initFromProperties();
    }

    public static RuntimeException exception(String msg, Object... args) {
        return asserter.exception(msg, args);
    }
}